overlord/snapstate/snapstate.go

```diff
@@ -703,6 +703,10 @@ func InstallWithDeviceContext(st *state.State, name string, opts *RevisionOption
 		return nil, err
 	}
 
+	if flags.RequireTypeBase && info.Type != snap.TypeBase && info.Type != snap.TypeOS {
+		return nil, fmt.Errorf("expected snap %q to be of type 'base'", name)
```

Comment:
thinking a bit further, maybe this one should be even more explicit

```
"declared snap base %q has unexpected type %q, instead of 'base'", name, info.Type  
```

?

Reply:
Good idea, done.